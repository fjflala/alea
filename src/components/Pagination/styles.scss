.ui-pagination {
  text-align: center;
  &__list {
    list-style: none;
  }

  &__item {
    display: inline-block;
    background-color: white;
    border: 1px solid #ccc;
    border-left: 0;

    &:first-child {
      border-left: 1px solid #ccc;
      border-top-left-radius: 4px;
      border-bottom-left-radius: 4px;
    }

    &:last-child {
      border-top-right-radius: 4px;
      border-bottom-right-radius: 4px;
    }

    button {
      background: transparent;
      border: 0;
      padding: 8px;
      outline: 0;
      font-size: 16px;
      cursor: pointer;
    }

    &--active {
      background-color: rgb(245,85,46);
      
      button {
        color: white;
      }
    }
  }
}