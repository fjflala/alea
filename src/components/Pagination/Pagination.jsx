/**
 * Module dependencies
 */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

/**
 * Import styles
 */
import './styles.scss';

/**
 * Represents a pagination
 * @param {object} props - Props of the element
 * @param {string} props.className - Custom className for the card
 * @param {number} props.totalPages - Total number of pages
 * @param {function} props.onChange - Callback when a element of the pagination is clicked
 */
const Pagination = ({
  totalPages,
  onChange,
  className,
}) => {
  const [pages, setPages] = useState([]);
  const [selectedPage, setSelectedPage] = useState(0);

  useEffect(() => {
    const quantityOfPages = totalPages && Array.from(Array(totalPages)).map((_, i) => i);
    setPages(quantityOfPages);
  }, [totalPages]);

  const onClickPage = (page) => {
    setSelectedPage(page);

    if (onChange && typeof onChange === 'function') {
      onChange(page);
    }
  };

  const cn = `ui-pagination ${className}`;

  return (
    <div className={cn}>
      <ol className="ui-pagination__list">
        {pages && pages.map(page => (
          <li
            key={page}
            className={`ui-pagination__item ${page === selectedPage ? 'ui-pagination__item--active' : ''}`}
          >
            <button
              type="button"
              onClick={() => onClickPage(page)}
            >
              {page + 1}
            </button>
          </li>
        ))}
      </ol>
    </div>
  );
};

/**
 * PropTypes definitions
 */
Pagination.propTypes = {
  totalPages: PropTypes.number,
  onChange: PropTypes.func,
  className: PropTypes.string,
};

Pagination.defaultProps = {
  totalPages: 0,
  onChange: () => {},
  className: '',
};

/**
 * Expose component
 */
export default Pagination;
