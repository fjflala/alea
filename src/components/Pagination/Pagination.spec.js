import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';
import { mount } from 'enzyme';

describe('<Pagination />', () => {
  let props;

  beforeEach(() => {
    props = {
      className: 'custom-class',
      onChange: jest.fn(),
      totalPages: 5,
    };
  });

  it('should correctly render', () => {

    const tree = renderer
      .create(<Pagination {...props} />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('should onChange prop', () => {
    const wrapper = mount(<Pagination {...props} />);

    wrapper.find('button').first().simulate('click');
    
    expect(props.onChange).toHaveBeenCalled();
    expect(props.onChange).toHaveBeenCalledWith(0);
  });
});