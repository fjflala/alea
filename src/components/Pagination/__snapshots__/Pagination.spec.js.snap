// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should correctly render 1`] = `
<div
  className="ui-pagination custom-class"
>
  <ol
    className="ui-pagination__list"
  />
</div>
`;
